### PR TITLE
Add compression_codec field to system.part_columns

### LINKED ISSUE #37311: Add compression_codec field to system.part_columns
**Use case**

When you apply `ALTER db.table MODIFY column CODEC(ZSTD(3))`, new codec applied only to new parts, or after background merges or after mutations. We need to watch the progress of codec changes.

**Describe the solution you'd like**

Add `compression_codec` field to system.part_columns, which will read from part
and use something like this

```
SELECT column, codec, active, count() parts 
FROM system.parts_columns 
WHERE database='titonissimo' AND table='request_log' 
AND column IN ('request_body','response_body') 
GROUP BY column, codec, active;
```
to watch progress

### PR DESCRIPTION
**Pull request by xlwh**
### Changelog category (leave one):
- New Feature

close #37311

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
...

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

### CODE REVIEW

File: src/Storages/System/StorageSystemPartsColumns.cpp [hunk 1]
```diff
@@ -182,6 +183,8 @@ void StorageSystemPartsColumns::processNextStorage(
                 columns[res_index++]->insert(index_size_in_bytes);
             if (columns_mask[src_index++])
                 columns[res_index++]->insert(index_size_in_allocated_bytes);
+            if (columns_mask[src_index++])
+                columns[res_index++]->insert(queryToString(part->default_codec->getCodecDesc()));
```
Comment (alesapin) [inline src/Storages/System/StorageSystemPartsColumns.cpp:187]: But now all part columns will have the same default compression codec. But we can specify codecs for columns separately, like:
```
CREATE TABLE compression_codec(
    id UInt64 CODEC(LZ4),
    data String CODEC(ZSTD),
    ddd Date CODEC(NONE),
    somenum Float64 CODEC(ZSTD(2)),
    somestr FixedString(3) CODEC(LZ4HC(7)),
    othernum Int64 CODEC(Delta)
) ENGINE = MergeTree() ORDER BY tuple();
```

**Review by alesapin — CHANGES_REQUESTED** [review]

File: src/Storages/System/StorageSystemPartsColumns.cpp (hunk 1 above)
Comment (xlwh) [reply src/Storages/System/StorageSystemPartsColumns.cpp:187]: Ok, I'll fix it later. thank you very much!

Comment (xlwh) [reply src/Storages/System/StorageSystemPartsColumns.cpp]: I have fixed the problem, please help and look again, thank you. @alesapin

File: src/Storages/System/StorageSystemPartsColumns.cpp [hunk 2]
```diff
@@ -94,7 +96,10 @@ void StorageSystemPartsColumns::processNextStorage(
             column_info.default_kind = toString(column.default_desc.kind);
             column_info.default_expression = queryToString(column.default_desc.expression);
         }
-
+        if (column.codec)
+        {
+            column_info.codec = queryToString(column.codec);
```
Comment (alesapin) [inline src/Storages/System/StorageSystemPartsColumns.cpp:101]: But we can change compression codecs with `ALTER`  queries. And obviously some parts can be compressed with old codec and some with new. We must handle it.

**Review by alesapin — CHANGES_REQUESTED** [review]

**Comment by alexey-milovidov** [comment]
Looks abandoned - no updates in three months.

**Pull request closed without merging**